openhands/integrations/forgejo/forgejo_service.py

```diff
@@ -0,0 +1,183 @@
+import os
+from typing import Any
+
+import httpx
+from pydantic import SecretStr
+
+from openhands.core.logger import openhands_logger as logger
+from openhands.integrations.service_types import (
+    AuthenticationError,
+    GitService,
+    ProviderType,
+    Repository,
+    UnknownException,
+    User,
+)
+from openhands.server.types import AppMode
+from openhands.utils.import_utils import get_impl
+
+
+class ForgejoService(GitService):
+    # Default to Codeberg, can be overridden
+    BASE_URL = os.environ.get('FORGEJO_BASE_URL', 'https://codeberg.org/api/v1')
```

Comment:
We allow for overriding the HOST via the provider token object 

Please take a look at how we pass this around in `openhands/integration/provider.py`